app/views/my/solutions/solve.html.haml

```diff
@@ -0,0 +1,105 @@
+.lo-container
+  #solve-interface
+    #ace-editor
+      :preserve
+        class TwoFer
+          def self.two_fer(name="you")
+            "One for %s, one for me." % name
+          end
+        end
+
+    #tests-output
+      .ops-status
+        Ops Status:
+        %span
+      .results-status
+        Tests Status:
+        %span
+      .results-message
+        Message:
+        %span
+      .test-failure
+        Failed test:
+        %span.name
+        .message
+
+  %button#run-tests-button.pure-button Run Tests
+
+:scss
+  #solve-interface {
+    display:flex;
+  }
+
+  #ace-editor {
+    width:50%;
+    height:300px;
+    margin-top:10px;
+    margin-bottom:10px;
+  }
+  #tests-output {
+    padding-left:50px;
+
+    .ops-status,
+    .results-status,
+    .results-message,
+    .test-failure {
+      display:none;
+    }
+  }
+  #run-tests-button {
+    background:green;
+  }
+
+-content_for :js do
```

Comment:
Yeah, totally. They would go in proper JS files now as well. I'm just trying not to pollute the main asset pipeline with this prototype, so that I don't have to think about it too much. Same for the `sass`, which IMO should never be compiled in the HAML/ERB as its very slow.